lib/ProductOpener/CRM.pm

```diff
@@ -0,0 +1,257 @@
+# This file is part of Product Opener.
+#
+# Product Opener
+# Copyright (C) 2011-2023 Association Open Food Facts
+# Contact: contact@openfoodfacts.org
+# Address: 21 rue des Iles, 94100 Saint-Maur des Fossés, France
+#
+# Product Opener is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Affero General Public License as
+# published by the Free Software Foundation, either version 3 of the
+# License, or (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Affero General Public License for more details.
+#
+# You should have received a copy of the GNU Affero General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+
+=encoding UTF-8
+
+=head1 NAME
+
+ProductOpener::CRM - manages integration with the Odoo CRM
+
+=head1 SYNOPSIS
+
+C<ProductOpener::CRM> contains functions to interact with the Odoo CRM
+
+
+=head1 DESCRIPTION
+
+[..]
+
+=cut
+
+package ProductOpener::CRM;
+
+use ProductOpener::PerlStandards;
+use Exporter qw< import >;
+
+BEGIN {
+	use vars qw(@ISA @EXPORT_OK %EXPORT_TAGS);
+	@EXPORT_OK = qw(
+		&create_company
+		&create_contact
+		&add_contact_to_company
+		&create_opportunity
+		&test
+	);
+	%EXPORT_TAGS = (all => [@EXPORT_OK]);
+
+}
+use vars @EXPORT_OK;
+
+use ProductOpener::Tags qw(%country_codes_reverse);
+use XML::RPC;
+use Log::Any qw($log);
+use Data::Dumper;
+use feature 'say';
+
+my %odoo_tags = (
+	'Producter' => '10',
+);
+
+our @api_credentials;
+our $xmlrpc;
+
+
+=head1 FUNCTIONS
+
+=head2 create_company ($org_ref)
+
+Creates a new company in odoo from an org
+
+=head3 Arguments
+
+=head4 $org_ref
+
+=head3 Return values
+
+the id of the created company
+
+=cut
+
+sub create_company ($org_ref) {
+	my $company = {
+		name => 		$org_ref->{name}, 
+		x_off_org_id => $org_ref->{org_id}, 
+		phone => 		$org_ref->{phone},
+		email => 		$org_ref->{email}, 
+		website => 		$org_ref->{website}, 
+		category_id =>  [$odoo_tags{Producter}],  # "Producter" category id in odoo
+		is_company => 	1,
+	};
+	my $company_id = odoo('res.partner', 'create', [{%$company}]);
+	$log->debug("create_company", {company_id => $company_id}) if $log->is_debug();
+	return $company_id;
+}
+
+
+
+=head2 create_contact ($user_ref)
+
+Creates a new contact in odoo from a user
+
+=head3 Arguments
+
+=head4 $org_ref
+
+=head3 Return values
+
+the id of the created contact
+
+=cut
+
+sub create_contact ($user_ref) {
```

Comment:
I imagine it comes with another PR but it would be important  to check user existence for the email in odoo.

Duplicates on contact are really annoying to the producers team.